frontend/src/components/utils/autoRefreshHandler.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+
+export default (
+  WrappedComponent: React.ComponentType<any>,
+  fetchDataFn: Function,
+  props?: any
+) => {
+  return class extends React.Component {
+    constructor(props: any) {
+      super(props);
+      this.timer = null;
+    }
+
+    timer: ReturnType<typeof setTimeout> | null;
+
+    state = {
+      items: []
+    };
+
+    componentDidMount() {
+      this.timer = setTimeout(this.regularFetchInfo, 0);
+    }
+
+    componentWillUnmount() {
+      clearTimeout(this.timer!);
+      this.timer = null;
```

Comment:
To avoid a race condition when we clear the timeout during `regularFetchInfo` execution we should write it like this:

```ts
const timer = this.timer;
this.timer = null;
if (timer !== null) {
  clearTimeout(timer);
}
```

Imagine the following situation:

1. `clearTimeout` while `regularFetchInfo` is executing
2. `regularFetchInfo` check for `this.timer !== null` and it is not `null` yet
3. `this.timer = null`
4. back to `regularFetchInfo` setting `this.timer = setTimeout(...)`

It is either super rare or even not possible in JS runtime, but I still feel uneasy about the race possibility (timers are the number one reason web pages become resources hogs).